applications/minotari_app_grpc/src/authentication/basic_auth.rs

```diff
@@ -118,12 +128,21 @@ impl BasicAuthCredentials {
 
         // Add the username bytes to the buffer
         let bytes_len_clipped = min(bytes.len(), MAX_USERNAME_LEN);
-        compare_bytes[0..bytes_len_clipped].clone_from_slice(&bytes[..bytes_len_clipped]);
+        compare_bytes
+            .get_mut(0..bytes_len_clipped)
+            .expect("Already checked")
+            .clone_from_slice(bytes.get(..bytes_len_clipped).expect("Already checked"));
```

Comment:
Just delete this line `let bytes_len_clipped = min(bytes.len(), MAX_USERNAME_LEN);` as well, then
```suggestion
        let (bytes_clipped, _suffix) = bytes.split_at(min(bytes.len(), MAX_USERNAME_LEN));
        let (prefix, suffix) = compare_bytes.split_at_mut(bytes_clipped.len());
        prefix.copy_from_slice(bytes_clipped);
```